### PR TITLE
Update C++ and Fortran example docstrings

### PR DESCRIPTION
**Pull request by speth**
This PR is a companion to Cantera/cantera-website#136.

**Changes proposed in this pull request**

- Standardize the header comments of C++/Fortran examples to make them play nice with the website
- Add missing license info to C++/Fortran examples
- A little bit of miscellaneous cleanup in these examples

**If applicable, fill in the issue number this pull request is fixing**

Fixes #

**Checklist**

- [x] There is a clear use-case for this code change
- [x] The commit message has a short title & references relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] The pull request is ready for review

### CODE REVIEW

**Review by 12Chao — COMMENTED** [review]
The commits look good to me, and I didn't find anything wrong.

File: samples/cxx/custom/custom.cpp [hunk 1]
```diff
@@ -1,7 +1,11 @@
 /*!
  * @file custom.cpp
- * Solve a closed-system constant pressure ignition problem where the governing equations
- * are custom-implemented.
+ *
+ * Custom Reactor
+ *
+ * Solve a closed-system constant pressure ignition problem where the governing
+ * equations are custom-implemented, using Cantera's interface to CVODES to
+ * integrate the equations.
```
Comment (jongyoonbae) [inline samples/cxx/custom/custom.cpp:8]: "Cantera's interface to CVODES to integrate" -> Do you mean "Cantera's interface CVODES to integrate"?

**Review by jongyoonbae — COMMENTED** [review]
Found one potential typo, but looks good otherwise.

File: samples/cxx/custom/custom.cpp (hunk 1 above)
Comment (bryanwweber) [reply samples/cxx/custom/custom.cpp:8]: Thanks @jongyoonbae! I think this is correct as written. Cantera provides an interfaces that connects to CVODES, which is actually does the numerical integration.

**Review by bryanwweber — APPROVED** [review]
Looks good, thanks @speth!

**Pull request merged**